src/openassetio-core/log/ConsoleLogger.cpp

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+#include <iomanip>
+#include <iostream>
+
+#include <openassetio/log/ConsoleLogger.hpp>
+
+// Foreground ANSI color codes (combined with \033[<color>m)
+constexpr const int kSeverityColors[] = {36, 32, 32, 0, 33, 31, 31};
+
+namespace openassetio {
+inline namespace OPENASSETIO_CORE_ABI_VERSION {
+namespace log {
+ConsoleLoggerPtr ConsoleLogger::make(bool colorOutput) {
+  return std::shared_ptr<ConsoleLogger>(new ConsoleLogger(colorOutput));
+}
+
+ConsoleLogger::ConsoleLogger(bool colorOutput) : colorOutput_(colorOutput) {}
+
+void ConsoleLogger::log(Severity severity, const Str& message) {
+  if (colorOutput_) {
+    std::cerr << "\033[0;" << kSeverityColors[severity] << "m";
```

Comment:
[consider] If we want to be grown-up, we should probably 
* Use `std::cerr` for `severity > kInfo` or possibly `> kWarning`.
* Use `std::clog` otherwise.

It's remarkably difficult to find easy to understand and detailed docs on this.  

`std::cerr` is unbuffered, so flushes with every `<<`, and it's also tied to `std:cout`, so forces a flush of `std::cout` before writing to stderr.

`std::clog`, on the other hand, also writes to stderr, but does not automatically flush and is not tied to `std::cout`.

So the general advice is, use `clog` for logging, `cerr` for errors - for performance when logging vs. immediate feedback when reporting errors.

Some optional references
* https://open-std.org/JTC1/SC22/WG21/docs/papers/2019/p1721r0.pdf (`29.4.3 Narrow stream objects`)
* https://en.cppreference.com/w/cpp/io/clog
* https://en.cppreference.com/w/cpp/io/cerr
* https://cplusplus.com/forum/general/284339/ - thread where people seem to know what they're talking about

Reply:
Deferring until #555.